lib/pghero/methods/space.rb

```diff
@@ -1,49 +1,128 @@
+        end
       end
 
       def relation_sizes
-        select_all_size <<-SQL
-          SELECT
-            n.nspname AS schema,
-            c.relname AS relation,
-            CASE WHEN c.relkind = 'r' THEN 'table' ELSE 'index' END AS type,
-            pg_table_size(c.oid) AS size_bytes
-          FROM
-            pg_class c
-          LEFT JOIN
-            pg_namespace n ON n.oid = c.relnamespace
-          WHERE
-            n.nspname NOT IN ('pg_catalog', 'information_schema')
-            AND n.nspname !~ '^pg_toast'
-            AND c.relkind IN ('r', 'i')
-          ORDER BY
-            pg_table_size(c.oid) DESC,
-            2 ASC
-        SQL
+        if !citus_enabled?
+          select_all_size <<-SQL
+            SELECT
+              n.nspname AS schema,
+              c.relname AS relation,
+              CASE WHEN c.relkind = 'r' THEN 'table' ELSE 'index' END AS type,
+              pg_table_size(c.oid) AS size_bytes
+            FROM
+              pg_class c
+            LEFT JOIN
+              pg_namespace n ON n.oid = c.relnamespace
+            WHERE
+              n.nspname NOT IN ('pg_catalog', 'information_schema')
+              AND n.nspname !~ '^pg_toast'
+              AND c.relkind IN ('r', 'i')
+            ORDER BY
+              pg_table_size(c.oid) DESC,
+              2 ASC
+          SQL
+        else
+          select_all_size <<-SQL
+            WITH dist_indexes_sizes AS ( 
+              WITH dist_indexes(idxname, tablename) AS ( 
+                SELECT 
+                  indexrelid, 
+                  indrelid 
+                FROM   
+                  pg_index 
+                JOIN   
+                  pg_dist_partition ON (indrelid = logicalrelid)
+              ), 
+              dist_indexes_shard_sizes AS ( 
+                SELECT 
+                  idxname::regclass, 
+                  (run_command_on_placements(tablename::regclass, $$
+                    SELECT 
+                      pg_table_size('$$ || idxname::regclass::text || $$' || replace('%s','$$ || tablename::regclass::text || $$', '')) 
+                  $$)).RESULT
+                FROM   
+                  dist_indexes 
+              ) 
+              SELECT   
+                idxname, 
+                SUM(RESULT::bigint) AS idxsize
+              FROM     
+                dist_indexes_shard_sizes 
+              GROUP BY 
+                idxname 
+            ) 
+            SELECT    
+              n.nspname AS SCHEMA, 
+              c.relname AS relation, 
+              CASE WHEN c.relkind = 'r' THEN 'table' ELSE 'index' END AS type, 
+              CASE WHEN c.relkind = 'r' THEN 
+                CASE WHEN ((SELECT COUNT(*) FROM pg_dist_partition WHERE logicalrelid = c.oid) > 0) THEN citus_table_size(c.oid) ELSE pg_table_size(c.oid) END 
```

Comment:
I didn't yet fully examined this query but always use EXISTS instead of COUNT(*) if you are only looking for existence.